lib/bricks-net-http-server.js

```diff
@@ -0,0 +1,295 @@
+				// Call the destructor-like method when the handler has finished.
+				connection.destroy();
+			});
+			
+			_accept();
+		}, function (err) {
+			_this._DEBUG_LOG(_this + ': Accept error:', err);
+			
+			_accept();
+		});
+	}
+	
+	when(_this.socket_).then(function () {
+		_accept();
+	}).done(function () {
+		_this.resolver.resolve(_this);
+	}, function (err) {
+		_this.resolver.reject(err);
+	});
+}
+HTTPServer.prototype.Register = function () {
+	var _this = this;
+	
+	cppArguments.assert('HTTPServer#Register', [
+		[
+			cppArguments.assertion('string', 'const std::string&', 'path'),
+			cppArguments.assertion(function (value) {
+				return (typeof value === 'function' && value.length === 1);
+			}, 'std::function<void(Request)>', 'handler'),
+			function (path, handler) {
+				if (_this.handlers_[path]) {
+					throw HandlerAlreadyExistsException(path);
+				}
+				_this.handlers_[path] = handler;
+				_this.handlers_paths_.push(path);
+			}
+		]
+	], arguments);
+	
+	return _this;
+};
+HTTPServer.prototype.UnRegister = function () {
+	var _this = this;
+	
+	cppArguments.assert('HTTPServer#UnRegister', [
+		[
+			cppArguments.assertion('string', 'const std::string&', 'path'),
+			function (path) {
+				if (!_this.handlers_[path]) {
+					throw HandlerDoesNotExistException(path);
+				}
+				delete _this.handlers_[path];
+				var index = _this.handlers_paths_.indexOf(path);
+				_this.handlers_paths_.splice(index, 1);
+			}
+		]
+	], arguments);
+	
+	return _this;
+};
+HTTPServer.prototype.ServeStaticFilesFrom = function () {
+	return cppArguments.assert('HTTPServer#ServeStaticFilesFrom', [
+		[
+			cppArguments.assertion('string', 'const std::string&', 'dir'),
+			cppArguments.assertion('string', 'const std::string&', 'route_prefix', cppArguments.ASSERTION_MODE_OPTIONAL),
+			function (dir, route_prefix) {
+				var _this = this;
+				
+				if (typeof route_prefix === 'undefined') {
+					route_prefix = "/";
+				}
+				
+				fs.readdirSync(dir).forEach(function (file) {
+					var content_type = GetFileMimeType(file, "");
+					if (content_type) {
+						_this.Register(
+							route_prefix + file,
+							new StaticFileServer(fs.readFileSync(path.join(dir, file)), content_type));
+					}
+					else {
+						throw CannotServeStaticFilesOfUnknownMIMEType(file);
+					}
+				});
+			}
+		]
+	], arguments, this);
+};
+HTTPServer.prototype.ResetAllHandlers = function () {
+	for (var ic = this.handlers_paths_.length, i = 0; i < ic; ++i) {
+		this.UnRegister(this.handlers_paths_[i]);
+		--i;
+		--ic;
+	}
+};
+HTTPServer.prototype.HandlersCount = function () {
+	return this.handlers_paths_.length;
+};
+
+
+var _runningServers = {};
+
+
+exports.HandlerAlreadyExistsException = HandlerAlreadyExistsException;
+exports.HandlerDoesNotExistException = HandlerDoesNotExistException;
+
+exports.ConnectionClose = ConnectionClose;
+exports.ConnectionKeepAlive = ConnectionKeepAlive;
+
+// Export to add the documentation in `bricks-net-api`.
```

Comment:
Fixed in https://github.com/sompylasar/KnowSheet-REPL/commit/348849574f127796424f00fc0bb1cd37f9160d5b